### PR TITLE
 v07-Theta prepare

### LINKED ISSUE #1215: Theta Release Meta Issue
Theta release meta issue for tracking all issues to make the release. Release should include:
* IBC v3 (confirm timelines, will need to work off of alpha for the time being)
  * Interchain Accounts (has its own meta issue)
* Cosmos SDK v0.45
  * confirm there are no breaking changes
* Liquidity module update
* Routing Module Update
  * strangelove said they'd begin on this
  * should confirm as well which version of IBC to target

### PR DESCRIPTION
**Pull request by yaruwangway**
Closes: #1215

- [x] bump cosmos-sdk to v0.45.1
- [x] bump ibc-go to v3
- [x] liquidity 1.5.0
- [x] check orders  in `SetOrder{BeginBlocker, EndBlocker, InitGenesis, ExportGenesis}` and check all modules are included
- [x] add migration logs when upgrade
- [x] test migration logs: "adding a new module: interchainaccounts" is in the log.
- [x] bump liquidity module
- [x] bump router module
- [ ] changelog
- [ ] check replace in go.mod
- [x] gaia module v6 -> v7
- [x] add Interchain Accounts

### CODE REVIEW

File: app/app.go
Comment (yaruwangway) [inline app/app.go:517]: Hi @robert-zaremba, @alexanderbez , does the order of`moduleNames` in `SetOrderBeginBlockers`, `SetOrderEndBlockers ` and `SetOrderInitGenesis ` functions matter ?

should `auth` comes after `bank` and `staking` in `SetOrderInitGenesis ` function ?
Thank you!

File: app/app.go
Comment (okwme) [inline app/app.go]: why did this change?

Comment (yaruwangway) [reply app/app.go]: before, the `transferModule` satisfy all the interface of `transfermodule` and `ibcmodule`, now these two are separate

**Comment by yaruwangway** [comment]
The migration log is not show during migrate due to the modules are the same version as before upgrade.

**Comment by okwme** [comment]
missing interchain accounts from IBC. Can take example of what's needed from our previous integration: 
https://github.com/cosmos/gaia/blob/ica-acct-auth/app/app.go

**Comment by yaruwangway** [comment]
@okwme last time you mentioned router module (packet-forward-middleware) is under some tests? how is the result ?

**Comment by yaruwangway** [comment]
I think we should not merge this to main, because if we have a new release process, we might want to rebase this branch or move it to other branches before merge to main.

File: app/app.go
Comment (yaruwangway) [inline app/app.go]: before, we block all module account addresses to receive tokens by `MsgSend`, but interchain account
 do not block distribution module (which scaffold from starport), should we keep blocking all the address or make distribution an exception ? 

https://github.com/cosmos/interchain-accounts/blob/7a4ade49c4ddc74b44e23c44906be371d00a322b/app/app.go#L547
https://github.com/cosmos/interchain-accounts/blob/7a4ade49c4ddc74b44e23c44906be371d00a322b/app/app.go#L168

Comment (okwme) [reply app/app.go]: we actually allow all modules in gaia https://github.com/cosmos/gaia/blob/v6.0.3/app/app.go#L653-L661

File: app/app.go
Comment (okwme) [reply app/app.go:517]: is this still an open question?

Comment (amaury1093) [reply app/app.go:517]: > does the order ofmoduleNames in SetOrderBeginBlockers, SetOrderEndBlockers and SetOrderInitGenesis functions matter ?

Yes. Unfortunately the easiest way currently to know the order is to follow the comments.

**Review by amaury1093 — APPROVED** [review]
SDK code lgtm (Set*Order functions, upgrade handler)

File: app/app.go
Comment (crodriguezvega) [inline app/app.go]: You can actually delete this line because this run already in a previous upgrade, right?

File: app/app.go
Comment (crodriguezvega) [inline app/app.go]: For ibc-go v3 upgrade you need to have here the following:

```
storeUpgrades := store.StoreUpgrades{
	Added: []string{icatypes.ModuleName},
}
```

To ensures that the new module's stores are added to the multistore before the migrations begin.

**Review by crodriguezvega — CHANGES_REQUESTED** [review]

File: app/app.go
Comment (damiannolan) [inline app/app.go]: You can remove this keeper. This comes from the `intertx` demo module for interchain-accounts.

File: app/app.go
Comment (crodriguezvega) [inline app/app.go:631]: When testing the upgrade of ibc-go from v2 to v3 with gaia, we found an issue and we have fixed it now in ibc-go v3.0.0-rc1. It's documented [here](https://github.com/cosmos/ibc-go/blob/v3.0.0-rc1/docs/migrations/v2-to-v3.md#upgrade-proposal). You will need to update your dependency. 
If you have any questions about it, let me know.

File: app/app.go
Comment (colin-axner) [reply app/app.go]: Won't allowing the distribution module to receive tokens break [invariants](https://github.com/cosmos/cosmos-sdk/blob/26d573e3dab889bb5527a454b1a6437a8760c263/x/distribution/keeper/invariants.go#L138)? While not well documented, this was the intention of the blocked addresses, to prevent invariants from being broken

File: app/app.go
Comment (crodriguezvega) [inline app/app.go]: Maybe for more clarity you could name this `transferIBCModule`.

File: app/app.go
Comment (crodriguezvega) [inline app/app.go]: You're using here the IBC transfer module as the ICA authentication module... It's a bit strange, but I guess you need to pass something here to make the compiler happy. Just keep in mind that you need to update this when you have a proper authentication module.

**Review by crodriguezvega — APPROVED** [review]
Great work, @yaruwangway! 🚀

File: go.mod
Comment (mmulji-ic) [inline go.mod:16]: @tychoish Any idea why this was reverted?

File: app/app.go
Comment (crodriguezvega) [inline app/app.go:643]: Just suggestion, so feel free to ignore it... Instead of hardcoding these strings you could do something like this: for example, for the `MsgSend` of bank, you can do `sdk.MsgTypeURL(&banktypes.MsgSend{})`, which should return the same string. This way you don't take the risk that you may have a typo in any of this harcoded strings.

Comment (glnro) [reply app/app.go:643]: +1 we should try to reduce any hardcoding wherever we can

**Pull request closed without merging**
